sk_sidekick/lib/src/commands/release_command.dart

```diff
@@ -0,0 +1,299 @@
+
+${cyan('Please open NEXT_RELEASE_CHANGELOG.md with the editor of your choice and edit it according to the instructions.')}
+''');
+
+    while (initialChangelog == nextReleaseChangelog.readAsStringSync()) {
+      sleep(1);
+    }
+
+    while (!confirm(
+      'Do you want to continue the release process with the changelog in NEXT_RELEASE_CHANGELOG.md?',
+      defaultValue: false,
+    )) {}
+
+    return nextReleaseChangelog;
+  }
+
+  String? _getMinSidekickCoreVersion(String pubspec) {
+    final doc = loadYamlDocument(pubspec);
+    final ps = doc.contents.value as YamlMap;
+    final constraint = ps['dependencies']['sidekick_core'] as String?;
+    if (constraint == null) {
+      return null;
+    }
+    return VersionConstraint.parse(constraint).minVersion.toString();
+  }
+
+  List<String>? _getSidekickCoreChanges(DartPackage package) {
+    final currentSidekickCoreVersion =
+        _getMinSidekickCoreVersion(package.pubspec.readAsStringSync());
+    if (currentSidekickCoreVersion != null) {
+      final currentSidekickCoreVersionTag =
+          'sidekick_core-v$currentSidekickCoreVersion';
+
+      final oldPubspecContent =
+          'git show $currentSidekickCoreVersionTag:${package.name}/pubspec.yaml'
+              .start(progress: Progress.capture())
+              .lines
+              .join('\n');
+      final oldSidekickCoreVersion =
+          _getMinSidekickCoreVersion(oldPubspecContent);
+
+      if (oldSidekickCoreVersion != null &&
+          currentSidekickCoreVersion != oldSidekickCoreVersion) {
+        final oldSidekickCoreVersionTag =
+            'sidekick_core-v$oldSidekickCoreVersion';
+        final newSidekickCoreCommits = _getChanges(
+          from: oldSidekickCoreVersionTag,
+          to: currentSidekickCoreVersionTag,
+          paths: ['sidekick_core'],
+        );
+
+        final message = '\nAlso package:sidekick_core updated '
+            '($oldSidekickCoreVersionTag -> $currentSidekickCoreVersionTag), '
+            'please consider those changes as well.\n';
+
+        return [message, ...newSidekickCoreCommits];
+      }
+    }
+    return null;
+  }
+}
+
+bool gitRepoHasLocalChanges(Directory directory) => 'git status --porcelain'
+    .start(progress: Progress.capture(), workingDirectory: directory.path)
+    .lines
+    .isNotEmpty;
+
+/// Returns all changes in the history from [from] (default: HEAD) to [to]
+/// If [paths] is given, returns only commits which modified [paths]
+///
+/// The format of each line is '<commit hash> <commit title>'
+List<String> _getChanges({
+  required String from,
+  String? to = 'HEAD',
+  List<String> paths = const [],
+}) =>
+    // %H = commit hash, %b = commit title
+    "git log --format='%H %s' $from..$to -- ${paths.join(' ')}"
+        .start(progress: Progress.capture())
+        .lines;
+
+void _warnIfNotOnDefaultBranch(Directory directory) {
+  final path = directory.path;
+  final defaultBranch = 'git rev-parse --abbrev-ref origin/HEAD'
```

Comment:
throws for me 
<img width="652" alt="Screen-Shot-2023-01-05-00-56-22 80" src="https://user-images.githubusercontent.com/1096485/210672033-09aa16fe-ed5e-45fb-8c7e-904d6d75bda5.png">

Can we simplify this by just checking for `git symbolic-ref HEAD` == `main`